### PR TITLE
Use MGet to query multi keys in RocksDB KV

### LINKED ISSUE #24364: [Enhancement]: Use MGET to load multi keys in RocksDB KV
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What would you like to be added?

As the title said, we can use MGET to load multi keys in RocksDB KV and this may bring performance benefits.

### Why is this needed?

RocksDB can perform optimizations such as async-io by io_uring. This would benefit the performance a lot when keys are lying in different levels of the LSM tree.

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by git-hulk**
RocksDB can perform optimizations such as async-io by io_uring. This would benefit the performance a lot when keys are lying in different levels of the LSM tree. 

This closes #24364

### CODE REVIEW

**Comment by jiaoew1991** [comment]
@git-hulk Welcome !

File: internal/kv/rocksdb/rocksdb_kv.go [hunk 1]
```diff
@@ -199,19 +199,13 @@ func (kv *RocksdbKV) LoadBytesWithPrefix(prefix string) ([]string, [][]byte, err
 
 // MultiLoad load a batch of values by keys
 func (kv *RocksdbKV) MultiLoad(keys []string) ([]string, error) {
-	if kv.DB == nil {
-		return nil, errors.New("rocksdb instance is nil when do MultiLoad")
+	valuesInBytes, err := kv.MultiLoadBytes(keys)
```
Comment (weiliu1031) [inline internal/kv/rocksdb/rocksdb_kv.go:202]: prefer to implement this with `kv.DB.MultiGet()`, cause call `kv.MultiLoadBytes()` directly which will create much more temporary  bytes objects

Comment (git-hulk) [reply internal/kv/rocksdb/rocksdb_kv.go:202]: @weiliu1031 Yes, the code author should also have this concern. But the original implementation also converted the byte array to the string, so I think it's fine to remove the duplicate code.

Comment (weiliu1031) [reply internal/kv/rocksdb/rocksdb_kv.go:202]: 1. notice origin line-223 in `kv.MultiLoadBytes()`, it create a temporary bytes object as return value. 
image

2. in `kv.MultiLoad()`, we can call `kv.DB.MultiGet()` directly, and then convert `value.Data()` to string, which will avoid the non-necessary temporary bytes in `kv.MultiLoadBytes()`

Comment (git-hulk) [reply internal/kv/rocksdb/rocksdb_kv.go:202]: @weiliu1031 Thanks for your information, I got your point now. So do you think I should revert this change?

Comment (weiliu1031) [reply internal/kv/rocksdb/rocksdb_kv.go:202]: > @weiliu1031 Thanks for your information, I got your point now. So do you think I should revert this change?

this pr is still in hold status, you can update your commit, and we will merge it, it a good job!

Comment (git-hulk) [reply internal/kv/rocksdb/rocksdb_kv.go]: done

**Comment by weiliu1031** [comment]
@git-hulk as mentioned above, please merge your commits into one commit

**Comment by git-hulk** [comment]
@weiliu1031 done

**Comment by git-hulk** [comment]
@weiliu1031 @xiaofan-luan 

For the CI failure, I'm not sure if it's related to the current change. The integration test was caused by the data race and I guess it's no related, but I cannot tell if Go unit test was related: https://pipelines.actions.githubusercontent.com/serviceHosts/c158e734-3f62-428e-85f8-58d396bf553e/_apis/pipelines/1/runs/146029/signedlogcontent/20?urlExpires=2023-05-25T14%3A40%3A54.1531149Z&urlSigningMethod=HMACV1&urlSignature=PXBg23vWeDE4nitjsKuM8UtNzob%2FHmM0ulGKJeDHZv8%3D

**Comment by weiliu1031** [comment]
> @weiliu1031 @xiaofan-luan
> 
> For the CI failure, I'm not sure if it's related to the current change. The integration test was caused by the data race and I guess it's not related, but I cannot tell if the Go unit test was related via the output.

go ut failed due to unstable ut case, we will fix it, you can retry your ut with `rerun ut`

**Comment by git-hulk** [comment]
@weiliu1031 thanks

**Comment by jiaoew1991** [comment]
@git-hulk Thanks for your contribution

**Pull request merged**